src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java

```diff
@@ -554,6 +555,18 @@ public Builder sorted()
             return this;
         }
 
+        /**
+         * Defines the maximum SSTable size when using the sorted writer.
+         * By default, i.e. not specified, there is no maximum size limit for the produced SSTable
+         * @param size the maximum size of each individual SSTable allowed
+         * @return this builder
+         */
+        public Builder withMaxSSTableSizeInMiBForSorted(int size)
+        {
+            this.maxSSTableSizeInMiBForSorted = size;
```

Comment:
we should do a validation for size , it should be positive, if a negative number is set , then we should remind users explicitly.
If not set then only one sstable will generate, and it's meaningless to set a negative or add some description to make a clear statement that negative number will generate only one sstable.

Reply:
Good point. I would prefer to log a warning here and let the write process to proceed, since producing a single sstable is a valid behavior, but might not be expected. How about this message?

"Maximum SSTable size is specified for the sorted CQLSSTableWriter, but the value is negative, which disables the size limiting effectively. Please supply a positive value in order to enforce size limiting for the produced SSTables."

Reply:
ok